app/controllers/internal/configs_controller.rb

```diff
@@ -3,14 +3,21 @@ class Internal::ConfigsController < Internal::ApplicationController
 
   before_action :extra_authorization_and_confirmation, only: [:create]
 
+  SOCIAL_MEDIA_PLATFORMS = %i[twitter facebook github instagram twitch]
+
   def show
+    @social_media_platforms = SOCIAL_MEDIA_PLATFORMS
     @logo_svg = SiteConfig.logo_svg.html_safe # rubocop:disable Rails/OutputSafety
   end
 
   def create
     clean_up_params
     config_params.keys.each do |key|
-      SiteConfig.public_send("#{key}=", config_params[key].strip) unless config_params[key].nil?
+      if key == "social_media_handles"
+        SiteConfig.public_send("#{key}=", config_params[key].to_h) unless config_params[key].empty?
```

Comment:
social_media_handles is a hash which needs to be dealt with differently to the rest